src/jdk.javadoc/share/classes/jdk/javadoc/doclet/package-info.java

```diff
@@ -312,20 +273,50 @@
  *         return Set.of(options);
  *     }
  *
- *     @Override
- *     public SourceVersion getSupportedSourceVersion() {
- *         // support the latest release
- *         return SourceVersion.latest();
- *     }
- * }
+ *    @Override
+ *    public SourceVersion getSupportedSourceVersion() {
+ *        return SourceVersion.latest();
+ *    }
+ *
+ *    @Override
+ *    public boolean run(DocletEnvironment environment) {
+ *        treeUtils = environment.getDocTrees();
+ *        new ShowFileObjects(reporter).show(environment.getSpecifiedElements());
+ *        return OK;
+ *    }
+ *
+ *    // A scanner to loop through specified elements
+ *    class ShowFileObjects {
+ *        final Reporter out;
+ *
+ *        ShowFileObjects(Reporter out) {
+ *            this.out = out;
+ *        }
+ *
+ *        void show(Set<? extends Element> elements) {
+ *            for (Element element : elements) {
+ *                if (element instanceof TypeElement te) {
```

Comment:
It is generally discouraged to use `instanceof` on `Element`s -- use `getKind` and a cast instead.
(yes, `instanceof` works on the JDK implementation but may not work on other implementations)